quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/providers/QuarkusConfigSourceProvider.java

```diff
@@ -24,20 +42,174 @@
+		collectStaticConfigSources(quarkusConfigSources, project);
+		collectProfileConfigSources(quarkusConfigSources, project);
+		return quarkusConfigSources;
+	}
+
+	private static void collectStaticConfigSources(List<IConfigSource> quarkusConfigSources, IJavaProject project) {
+		quarkusConfigSources.add(new YamlConfigSource(APPLICATION_YAML_FILE, project));
+		quarkusConfigSources.add(new YamlConfigSource(APPLICATION_YML_FILE, project));
+		quarkusConfigSources.add(new PropertiesConfigSource(APPLICATION_PROPERTIES_FILE, project, 250));
+	}
+
+	private void collectProfileConfigSources(List<IConfigSource> quarkusConfigSources, IJavaProject project) {
+		if (!projectProfileConfigSources.containsKey(project)) {
+			File classesFolder = getClassesFolder(project);
+			initializeProfileConfigSources(project, classesFolder);
+			JavaCore.addPreProcessingResourceChangedListener(
+					new ApplicationProfileConfigSourceChangeListener(project, projectProfileConfigSources.get(project), classesFolder),
+					IResourceChangeEvent.POST_CHANGE | IResourceChangeEvent.PRE_DELETE);
+		}
+		synchronized (projectProfileConfigSources.get(project)) {
+			List<IConfigSource> profileConfigSources = projectProfileConfigSources.get(project);
+			quarkusConfigSources.addAll(profileConfigSources);
+		}
+	}
+
+	private void initializeProfileConfigSources(IJavaProject project, File classesFolder) {
+		List<IConfigSource> profileConfigSources = new ArrayList<>();
+		for (File file : classesFolder.listFiles()) {
+			if (file.isFile()) {
+				Matcher m = PER_PROFILE_FILE_NAME_PTN.matcher(file.getName());
+				if (m.matches()) {
+					// Not synchronized, since the resource change listener should not be started
+					// yet
+					profileConfigSources.add(
+							new ApplicationProfileConfigSource(file.getName(), m.group(1), project));
+				}
+			}
+		}
+		projectProfileConfigSources.put(project, profileConfigSources);
+	}
+
+	/**
+	 * Returns the folder where the classes are placed for the given project and null if there is none
+	 *
+	 * @param project the project to find the resources folder for
+	 * @return the folder where the classes are placed for the given project and null if there is none
+	 */
+	private static File getClassesFolder(IJavaProject project) {
+		try {
+			for (IClasspathEntry sourceEntry : ((JavaProject) project).getResolvedClasspath(true)) {
+				if (!sourceEntry.isTest() && sourceEntry.getEntryKind() == IClasspathEntry.CPE_SOURCE
+						&& sourceEntry.getOutputLocation() != null) {
+					IPath output = sourceEntry.getOutputLocation();
+					File outputFile = project.getProject().getLocation().append(output.removeFirstSegments(1)).toFile();
+					if (outputFile.exists() && outputFile.isDirectory()) {
+						return outputFile;
+					}
+				}
+			}
+		} catch (JavaModelException e) {
+			LOGGER.log(Level.WARNING, "Error while finding the resources folder", e);
+		}
+		return null;
+	}
+
+	private static class ApplicationProfileConfigSourceChangeListener
```

Comment:
This kind of listener should be done in LSP4MP. The idea is to cache all IConfigSource and update the list when there is a properties/yaml file which is deleted.

Could you please remove all code which is related to ApplicationProfileConfigSourceChangeListener

Reply:
https://github.com/eclipse/lsp4mp/issues/164